File: src/monkey/repl/repl.py
from monkey.lexer.lexer import Lexer

from monkey.tokens.tokens import Tokens


PROMPT = ">>> "


def start_repl():
    while inp := input(PROMPT):
        lexer = Lexer(inp)
        while tok := lexer.next_token():
            if tok.type == Tokens.EOF:
                break
            print(tok)

File: rust/src/repl.rs
use crate::lexer::Lexer;
use crate::token::Token;
use std::io::{stdin, Result, Write};

pub fn run_repl() -> Result<()> {
    print_prompt();
    for line in stdin().lines() {
        let input = line.unwrap();
        let mut lexer = Lexer::new(input);
        loop {
            let tok = lexer.next_token().unwrap();
            println!("{:?}", tok);
            if tok == Token::Eof {
                break;
            }
        }
        print_prompt();
    }
    Ok(())
}

fn print_prompt() {
    print!(">> ");
    std::io::stdout().flush().unwrap();
}
